src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -104,6 +97,25 @@ class LedgerImpl(vm: VM, blockchainConfig: BlockchainConfig) extends Ledger with
     blockTxsExecResult
   }
 
+
+  private[ledger] def executePreparedTransactions(
```

Comment:
Minor detail: As this function is tail recursive, maybe we could use the `@tailrec` annotation?